src/Http/Response.php

```diff
@@ -2387,6 +2387,20 @@ public function getCookieCollection()
         return $this->_cookies;
     }
 
+    /**
+     * Get a new instance with provided cookie collection.
+     *
+     * @param \Cake\Http\Cookie\CookieCollection $cookieCollection Cookie collection to set.
+     * @return static
+     */
+    public function withCookieCollection(CookieCollection $cookieCollection)
```

Comment:
Should this hint be for the interface?

Reply:
`CookieCollection` doesn't implement any interface besides `IteratorAggregate` and `Countable`.

Reply:
My bad. I thought we had interfaces.